### PR TITLE
Implement editor bottom toolbar

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
What's still needed to complete this?

File: src/OpenLoco/Graphics/ImageIds.h [hunk 1]
```diff
@@ -24,7 +24,8 @@ namespace OpenLoco::ImageIds
     constexpr uint32_t inline_red_down_arrow = 2325;
 
     constexpr uint32_t construction_straight = 2335;
-
+    constexpr uint32_t spr_2336 = 2336;
+    constexpr uint32_t spr_2337 = 2337;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Graphics/ImageIds.h:28]:
```suggestion
    constexpr uint32_t previous_step = 2336;
    constexpr uint32_t next_step = 2337;
```

Comment (marijnvdwerf) [reply src/OpenLoco/Graphics/ImageIds.h:28]: image

File: src/OpenLoco/EditorController.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    // 0x0043D7DC
+    void init()
+    {
+        call(0x0043D7DC);
+    }
+
+    Step getCurrentStep()
+    {
+        return (Step)*_editorStep;
```
Comment (duncanspumpkin) [inline src/OpenLoco/EditorController.cpp:20]: Why the cast? Can't you just return `_editorStep` that should handle all of the dereferencing.

File: src/OpenLoco/EditorController.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    Step getPreviousStep()
+    {
+        return (Step)((uint8_t)(*_editorStep) - 1);
```
Comment (duncanspumpkin) [inline src/OpenLoco/EditorController.cpp:25]: Use the constructor `Step(static_cast<uint8_t>(*_editorStep) - 1)`

File: src/OpenLoco/EditorController.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    Step getNextStep()
+    {
+        return (Step)((uint8_t)(*_editorStep) + 1);
```
Comment (duncanspumpkin) [inline src/OpenLoco/EditorController.cpp:30]: as above

File: src/OpenLoco/Windows/ToolbarBottomEditor.cpp [hunk 5]
```diff
@@ -1,15 +1,145 @@
...
+
+    // 0x0043CE65
+    static void draw(window* self, Gfx::drawpixelinfo_t* ctx)
+    {
+        widget_t& previous = self->widgets[widx::previous_frame];
+        widget_t& next = self->widgets[widx::next_frame];
+
+        if (EditorController::canGoBack())
+        {
+            Gfx::drawRect(ctx, previous.left + self->x, previous.top + self->y, previous.width(), previous.height(), 0x2000000 | 52);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Windows/ToolbarBottomEditor.cpp:68]: `Gfx::recolour(52); `

File: src/OpenLoco/Windows/ToolbarBottomEditor.cpp [hunk 6]
```diff
@@ -1,15 +1,145 @@
...
+        Gfx::drawRectInset(ctx, next.left + self->x + 1, next.top + self->y + 1, next.width() - 2, next.height() - 2, self->colours[1], 0x30);
+
+        Gfx::drawStringCentred(*ctx, (previous.right + next.left) / 2 + self->x, self->y + self->height - 12, Colour::opaque(self->colours[2]) | Colour::outline_flag, _stepNames[EditorController::getCurrentStep()]);
+
+        if (EditorController::canGoBack())
+        {
+            Gfx::drawImage(ctx, self->x + previous.left + 6, self->y + previous.top + 6, ImageIds::step_back);
+            int x = (previous.left + 30 + previous.right) / 2;
+            int y = previous.top + 6;
+            colour_t al = Colour::opaque(self->colours[1]);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Windows/ToolbarBottomEditor.cpp:87]: `textColour`

File: src/OpenLoco/Windows/ToolbarBottomEditor.cpp [hunk 7]
```diff
@@ -1,15 +1,145 @@
...
+            {
+                al = Colour::white;
+            }
+            Gfx::drawStringCentred(*ctx, self->x + x, self->y + y, al, StringIds::editor_previous_step);
+            Gfx::drawStringCentred(*ctx, self->x + x, self->y + y + 10, al, _stepNames[EditorController::getPreviousStep()]);
+        }
+        Gfx::drawImage(ctx, self->x + next.right - 29, self->y + next.top + 4, ImageIds::step_forward);
+        int x = next.left + (next.width() - 31) / 2;
+        int y = next.top + 6;
+        colour_t al = Colour::opaque(self->colours[1]);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Windows/ToolbarBottomEditor.cpp:98]: `textColour`

File: src/OpenLoco/Windows/ToolbarBottomEditor.cpp [hunk 8]
```diff
@@ -1,15 +1,145 @@
...
+        }
+    }
+
     // 0x0043CCCD
     void open()
     {
-        call(0x0043CCCD);
-        // Note call to window_resize_gui_scenario_editor not part of this function
-        // move when implemented.
+        events.on_mouse_up = onMouseUp;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Windows/ToolbarBottomEditor.cpp:125]: I'd still make a separate function for setting the events to just make it consistent with the rest of the codebase.

**Review by duncanspumpkin — APPROVED** [review]

**Review by AaronVanGeffen — COMMENTED** [review]
Echoing @duncanspumpkin's comments. The toolbar itself appears to be working well. Repositions nicely when resizing the window and all that.

When the scenario editor is closed, the scenario is saved properly. However, on the title screen, an error pops up immediately after:
Yew Island

(The file browser still suffers from #765, but I remedied that with a rebase so #780 was in the tree.)

**Comment by marijnvdwerf** [comment]
@AaronVanGeffen Same error appears on `master` for me.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**